server/svix-server/src/worker.rs

```diff
@@ -284,17 +374,46 @@ async fn dispatch(
         headers
     };
 
-    let client = reqwest::Client::builder()
-        .redirect(reqwest::redirect::Policy::none())
-        .build()
-        .expect("Invalid reqwest Client configuration");
-    let res = client
-        .post(&endp.url)
-        .headers(headers)
-        .timeout(Duration::from_secs(cfg.worker_request_timeout as u64))
-        .json(&payload)
-        .send()
-        .await;
+    // Creates the `HttpConnector` with overridden DNS resoluteon and a connection timeout
+    let mut connector = hyper::client::connect::HttpConnector::new_with_resolver(DnsResolver {
+        allowed_private_subnets: vec![],
+    });
+    connector.set_connect_timeout(Some(Duration::from_secs(cfg.worker_request_timeout as u64)));
```

Comment:
You don't want just a connect timeout, you want it to be the whole requset timeout.